File: Blog.Core/Configuration/BlogConfiguration.cs
namespace Blog.Core.Configuration
{
    public class BlogConfiguration
    {
        public bool DisplayFullErrorStack { get; set; }

        public string StaticFilesCacheControl { get; set; }

        public bool UseResponseCompression { get; set; }

        public string AzureBlobStorageConnectionString { get; set; }

        public string AzureBlobStorageContainerName { get; set; }

        public string AzureBlobStorageEndPoint { get; set; }

        public bool RedisCachingEnabled { get; set; }

        public string RedisCachingConnectionString { get; set; }

        public bool PersistDataProtectionKeysToRedis { get; set; }

        public string UserAgentStringsPath { get; set; }

        public string CrawlerOnlyUserAgentStringsPath { get; set; }

        public bool SupportPreviousNopcommerceVersions { get; set; }

        public bool DisableSampleDataDuringInstallation { get; set; }

        public bool UseFastInstallationService { get; set; }

        public string PluginsIgnoredDuringInstallation { get; set; }

        public bool IgnoreStartupTasks { get; set; }

        public bool ClearPluginShadowDirectoryOnStartup { get; set; }

        public bool CopyLockedPluginAssembilesToSubdirectoriesOnStartup { get; set; }

        public bool UseUnsafeLoadAssembly { get; set; }

        public bool UsePluginsShadowCopy { get; set; }
    }
}

File: Shared/Blog.FSharp.Core/Configuration/BlogConfiguration.fs
namespace Blog.Core.Configuration

/// Application-wide blog configurations.
type BlogConfiguration = {
    
    // Error & Response
    DisplayFullErrorStack: bool
    UseResponseCompression: bool
    StaticFilesCacheControl: string option

    // Azure Blob Storage
    AzureBlobStorageConnectionString: string option
    AzureBlobStorageContainerName: string option
    AzureBlobStorageEndPoint: string option

    // Redis
    RedisCachingEnabled: bool
    RedisCachingConnectionString: string option
    PersistDataProtectionKeysToRedis: bool

    // User Agent
    UserAgentStringsPath: string option
    CrawlerOnlyUserAgentStringsPath: string option

    // Installation
    SupportPreviousNopcommerceVersions: bool
    DisableSampleDataDuringInstallation: bool
    UseFastInstallationService: bool
    PluginsIgnoredDuringInstallation: string option

    // Plugins & Startup
    IgnoreStartupTasks: bool
    ClearPluginShadowDirectoryOnStartup: bool
    CopyLockedPluginAssembilesToSubdirectoriesOnStartup: bool
    UseUnsafeLoadAssembly: bool
    UsePluginsShadowCopy: bool
}